src/components/fullscreen/fullscreen.js

```diff
@@ -65,8 +67,50 @@ class FullscreenControl extends BaseComponent {
       typeof document.mozFullScreenElement !== 'undefined' && Boolean(document.mozFullScreenElement) ||
       typeof document.msFullscreenElement !== 'undefined' && Boolean(document.msFullscreenElement);
 
-    isFullscreen ? this.player.notifyEnterFullscreen() : this.player.notifyExitFullscreen();
-    this.props.updateFullscreen(isFullscreen);
+    isFullscreen ? this.fullscreenEnterHandler() : this.fullscreenExitHandler();
+  }
+
+  /**
+   * fullscreen enter handler function. updates the store with new value
+   *
+   * @returns {void}
+   * @memberof FullscreenControl
+   */
+  fullscreenEnterHandler(): void {
+    this.player.notifyEnterFullscreen();
+    this.props.updateFullscreen(true);
+  }
+
+  /**
+   * fullscreen exit handler function. updates the store with new value
+   *
+   * @returns {void}
+   * @memberof FullscreenControl
+   */
+  fullscreenExitHandler(): void {
+    this.player.notifyExitFullscreen();
+    this.props.updateFullscreen(false);
+  }
+
+  /**
+   * Attach listeners to ios full screen change
+   *
+   * @returns {void}
+   * @memberof FullscreenControl
+   */
+  attachIosFullscreenListeners(): void {
+    this.player.getVideoElement().addEventListener('webkitbeginfullscreen', () => {
```

Comment:
why not add these in `componentDidMount` and add them only if device is iOS and Tablet, or better yet, if device is iPad?

Reply:
`this.player.getVideoElement()` is `undefined` in `componentDidMount `. i have to wait for an engine.
The first triggered event once engine selected is SOURCE_SELECTED.